### PR TITLE
Phemex - editOrder implementation

### LINKED ISSUE #10073: Best way to edit existing order to add stop loss or trailing stop loss - phemex
what the best quickest way to edit an existing order/position?

### PR DESCRIPTION
**Pull request by ttodua**
Implemented editorder

(btw, might need revision whether my used `market['valueScale']` is the one that needs to be used there. Phemex had a bit complex structure & insufficient docs to find the explanation of all params, which I could used for creating editorder).

[ fixes #10073 ]

### CODE REVIEW

**Comment by ttodua** [comment]
@kroitor is there any problem with travis because of the repo size? it seems to frequently (maybe 2-3 times from 10 ) the build fails (i.e. `Receiving objects: 100% (386506/386506), 1.15 GiB | 39.66 MiB/s, done. ` and then after enumerating the tags it ends with '...No output has been received in the last 10m0s...'

I still think about whether it might be some good to archive the 'old' commit data or somehow reduce the repo size, if that slows down things,  idk...

File: js/phemex.js [hunk 1]
```diff
@@ -1922,6 +1923,49 @@ module.exports = class phemex extends Exchange {
...
+        params = this.omit (params, [ 'priceEp' ]);
+        if (finalPrice !== undefined) {
+            request['priceEp'] = this.priceToPrecision (symbol, finalPrice);
+        } else if (price !== undefined) {
+            request['priceEp'] = this.priceToPrecision (symbol, price * Math.pow (10, market['valueScale']));
+        }
+        const finalQty = this.safeString (params, 'baseQtyEV');
+        params = this.omit (params, [ 'baseQtyEV' ]);
+        if (finalQty !== undefined) {
+            request['baseQtyEV'] = this.amountToPrecision (symbol, finalQty);
```
Comment (kroitor) [inline js/phemex.js:1959]: I think it's `baseQtyEv`, with a lowercase `v`, no?

File: js/phemex.js [hunk 2]
```diff
@@ -1922,6 +1923,49 @@ module.exports = class phemex extends Exchange {
...
+        } else if (amount !== undefined) {
+            request['baseQtyEV'] = this.amountToPrecision (symbol, amount * Math.pow (10, market['valueScale']));
```
Comment (kroitor) [inline js/phemex.js:1961]: I suggest that we copy this logic from `createOrder` and use `.toEv`, `.toEp` exchange-specific methods that were implemented for this purpose.

File: js/phemex.js (hunk 1 above)
Comment (ttodua) [reply js/phemex.js:1959]: @kroitor strange thing, in API example REQUEST URL they have with uppercase: https://github.com/phemex/phemex-api-docs/blob/master/Public-Spot-API-en.md#amend-order

but in other places I see with lowercase. When I submitted PR i tested the code and it worked as I remind, but will re-check.

Comment (ttodua) [reply js/phemex.js:1959]: I've tested out and surprisingly, it is working only when 'baseQtyEV' (last 'v' uppercase), with lowercased 'v' request it doesn't work

Comment (kroitor) [reply js/phemex.js:1959]: @ttodua thank you!

**Comment by ttodua** [comment]
@kroitor PR ready now.

File: js/phemex.js [hunk 3]
```diff
@@ -1922,6 +1923,54 @@ module.exports = class phemex extends Exchange {
...
+        params = this.omit (params, [ 'clientOrderId', 'clOrdID' ]);
+        if (clientOrderId !== undefined) {
+            request['clOrdID'] = clientOrderId;
+        } else {
+            request['orderID'] = id;
+        }
+        const finalPrice = this.safeString (params, 'priceEp');
+        params = this.omit (params, [ 'priceEp' ]);
+        if (finalPrice !== undefined) {
+            request['priceEp'] = this.priceToPrecision (symbol, finalPrice);
```
Comment (kroitor) [inline js/phemex.js:1951]: I think, if the user has supplied `priceEp` – it's already formatted to precision, and we should not do `priceToPrecision` here.

Comment (kroitor) [reply js/phemex.js:1951]: The same goes to `quantity` below.

Comment (kroitor) [reply js/phemex.js:1951]: If we take the above into the account and project this further, then `finalPrice` should not be there at all.
Lines 1948-1954 could be simplified to:
```
if (price !== undefined) {
    request['priceEp'] = this.toEp (price, market);
}
```
The request will be extended with `params` before sending it out anyways, so a user-defined `priceEp` override would take precedence in any case.

Comment (ttodua) [reply js/phemex.js:1951]: At, totally right.

Comment (ttodua) [reply js/phemex.js]: Lmk what you think now. We also could remove `baseQtyEv` from manual defining approach too, but for the sake of users (as exchange has is erroneously uppercased) I thought support for lowercase would be good too.

**Pull request merged**